packages/adapter/addon/rest.ts

```diff
@@ -1369,7 +1369,9 @@ function fetchErrorHandler(
     payload = responseData.errorThrown.payload;
   } else {
     responseData.errorThrown = errorThrown;
-    payload = adapter.parseErrorResponse(payload as string);
+    if (typeof payload === 'string') {
```

Comment:
This seems a bit sketch. We probably want to always call `parseErrorResponse`

Reply:
The trouble I was coming across here is JSON.parse `undefined` (`Payload: string | undefined | null`).  I know we have the `catch` there but then JSON.parse will complain if we pass anything other than `string`.  Have any thoughts on how we can get around this?